reader.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func getFilesFromFolder(folder string) (files []string, err error) {
+	err = filepath.Walk(folder,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
```

Comment:
Erros mais descritivos irão facilitar a depuração

```suggestion
				return fmt.Errorf("Erro iterando no arquivo \"%s\". Diretório base \"%s\". Erro: %q", path, folder, err)
```